src/community/script/core/src/main/java/org/geoserver/script/rest/converter/ScriptConverter.java

```diff
@@ -0,0 +1,179 @@
+
+/**
+ * This class enables us to extend the XMLXStreamListConverter and JSONXStreamListConverter classes so we can
+ * override the default XML and JSON output of the REST module.  Because these message converters have a lower
+ * priority than the converters in the REST module, they will always be called and therefore to prevent affecting
+ * the expected behavior of the REST module we are checking the request path and only customizing the output
+ * if the request is within the Scripts module.
+ */
+public abstract class ScriptConverter extends XStreamCatalogListConverter {
+
+    /**
+     * XML handling for Script lists
+     */
+    @Component
+    public static class XMLXStreamScriptListConverter extends XMLXStreamListConverter {
+        @Autowired
+        HttpServletRequest request;
+
+        @Override
+        protected XStream createXStreamInstance() {
+            return new SecureXStream();
+        }
+
+        @Override
+        public void encodeLink(String link, HierarchicalStreamWriter writer) {
+            if (checkPath(request)) {
+                try {
+                    link = URLDecoder.decode(link, "UTF-8");
+                    encodeAlternateAtomLinkNoExt(link, writer);
+                    writer.addAttribute("type", getMediaType());
+
+                    writer.endNode();
+                } catch (UnsupportedEncodingException e) {
+
+                }
+            } else {
+                encodeAlternateAtomLink(link, writer);
+            }
+        }
+
+        @Override
+        public void encodeCollectionLink(String link, HierarchicalStreamWriter writer) {
+            if (checkPath(request)) {
+                try {
+                    link = URLDecoder.decode(link, "UTF-8");
+                    encodeAlternateAtomLinkNoExt(link, writer);
+                    writer.addAttribute("type", getMediaType());
+
+                    writer.endNode();
+                } catch (UnsupportedEncodingException e) {
+
+                }
+            } else {
+                encodeAlternateAtomLink(link, writer);
+            }
+        }
+
+        @Override
+        public String getMediaType() {
+            return MediaType.APPLICATION_ATOM_XML_VALUE;
+        }
+
+        @Override
+        public String getExtension() {
+            return "xml";
+        }
+    }
+
+    /**
+     * JSON handling for Script lists
+     */
+    @Component
+    public static class JSONXStreamScriptListConverter extends JSONXStreamListConverter {
+        @Autowired
+        HttpServletRequest request;
+
+        @Override
+        public void encodeLink(String link, HierarchicalStreamWriter writer) {
+            if (checkPath(request)) {
+                try {
+                    link = URLDecoder.decode(link, "UTF-8");
+                    writer.startNode("href");
+                    writer.setValue(hrefNoExt(link));
+                    writer.endNode();
+                } catch (UnsupportedEncodingException e) {
+
```

Comment:
You should at least log a WARNING or DEBUG message here, swallowing exceptions makes debugging unnecessarily difficult